### PR TITLE
Improve clean logic

### LINKED ISSUE #58: Include missfont.log in clean
When using the Clean command, the file `missfont.log` isn't removed.

I think that file should be removed too..

Thanks

### LINKED ISSUE #161: Clean does not clean subdirectories (using latexmk)
When including tex-files of subdirectories with

    \include{subdir/Chapter1}

and using the latexmk engine, at least all aux-files still remain in this subdirectory after cleaning.

To reproduce this, use my small test repo [orzechow/latexmk-clear-test](https://github.com/orzechow/latexmk-clear-test).

At first I thought it might be a latexmk bug and actually encoutered one with exactly the same behaviour. This latexmk bug has been confirmed, but as I saw in the code, atom-latex does not even call latexmk for cleaning. (Why not actually?)
So it seems to be an issue of atom-latex.

### PR DESCRIPTION
**Pull request by yitzchak**
This PR is attempt to improve the clean logic, make it more flexible and yet still remaining safe. The current logic is:

1. Get a list of files from the FDB and any files from the output directory that start with the job name. This becomes the `generatedFiles`.
2. Loop through the list of patterns in `cleanPatterns`.
3. Do named property replacement on each pattern. Currently, `{jobname}`, `{output}` or `{name}`.
4. If the pattern is an absolute path (`/minted-{jobname}`) use it as a glob on the file system after joining to the root path.
5. If the pattern is a relative path, (`**/*.aux`) use it as a glob pattern on `generatedFiles` after joining to the root path.
6. :zap: any matches from step 4 or step 5.

This resolves #58 and resolves #161

### CODE REVIEW

File: lib/composer.js [hunk 1]
```diff
@@ -170,6 +170,7 @@ export default class Composer {
 
     for (const pattern of patterns) {
       const absolutePattern = path.join(rootPath, pattern)
+      console.log(`${pattern} -> ${absolutePattern}`)
```
Comment (thomasjo) [inline lib/composer.js]: 🔥

File: lib/composer.js [hunk 2]
```diff
@@ -5,7 +5,9 @@ import { shell } from 'electron'
 import fs from 'fs-plus'
 import path from 'path'
 import BuilderRegistry from './builder-registry'
-import { getEditorDetails, heredoc } from './werkzeug'
+import { getEditorDetails, heredoc, replaceProperitiesInString } from './werkzeug'
```
Comment (thomasjo) [inline lib/composer.js]: Typo here; should be `replacePropertiesInString`.

File: lib/composer.js [hunk 3]
```diff
@@ -154,33 +153,80 @@ export default class Composer {
...
+    for (const file of files.values()) {
+      fs.removeSync(file)
     }
+  }
 
-    if (jobname) {
-      name = jobname
+  getCleanPatterns (builder, rootFilePath, jobname) {
+    const { name, ext } = path.parse(rootFilePath)
+    const output = atom.config.get('latex.outputDirectory')
```
Comment (thomasjo) [inline lib/composer.js]: I really think we should change `output` to `outputDir`. This applies to all other instances of `output`.

File: lib/werkzeug.js [hunk 4]
```diff
@@ -30,5 +30,9 @@ export default {
     const lineNumber = position.row + 1
 
     return { editor, filePath, position, lineNumber }
+  },
+
+  replaceProperitiesInString (text, properties) {
```
Comment (thomasjo) [inline lib/werkzeug.js]: Same typo here obviously; should be `replacePropertiesInString`.

File: lib/composer.js (hunk 2 above)
Comment (yitzchak) [inline lib/composer.js]: Nice catch.

**Pull request merged**